plugins/teststeps/sshcmd/sshcmd.go

```diff
@@ -186,31 +201,49 @@ func (ts *SSHCmd) Run(cancel, pause <-chan struct{}, ch test.TestStepChannels, p
 			errCh <- innerErr
 		}()
 
-		select {
-		case err := <-errCh:
-			log.Infof("Stdout of command '%s' is '%s'", cmd, stdout.Bytes())
-			if err == nil {
-				// Execute expectations
-				expect := ts.Expect.String()
-				if expect == "" {
-					log.Warningf("no expectations specified")
-				} else {
-					re := regexp.MustCompile(expect)
-					matches := re.FindAll(stdout.Bytes(), -1)
-					if len(matches) > 0 {
-						log.Infof("match for regex \"%s\" found", expect)
+		expect := ts.Expect.String()
+		for {
+			select {
+			case err := <-errCh:
+				log.Infof("Stdout of command '%s' is '%s'", cmd, stdout.Bytes())
+				if err == nil {
+					// Execute expectations
+					if expect == "" {
+						log.Warningf("no expectations specified")
 					} else {
-						return fmt.Errorf("match for %s not found for target %v", expect, target)
+						re := regexp.MustCompile(expect)
+						matches := re.FindAll(stdout.Bytes(), -1)
+						if len(matches) > 0 {
+							log.Infof("match for regex \"%s\" found", expect)
+						} else {
+							return fmt.Errorf("match for %s not found for target %v", expect, target)
+						}
 					}
+				} else {
+					log.Warningf("Stderr of command '%s' is '%s'", cmd, stderr.Bytes())
+				}
+				return err
+			case <-cancel:
+				return session.Signal(ssh.SIGKILL)
+			case <-pause:
+				return session.Signal(ssh.SIGKILL)
+			default:
+				re := regexp.MustCompile(expect)
```

Comment:
compile this once, after retrieving.
also, assuming this is supplied by the user, this will crash in case of a bad regex, which is bad.
MustCompile should only be used for static regexes.

Reply:
Done.